github.js

```diff
@@ -199,20 +148,14 @@ function configureCredentials(config, ui) {
 
   return Promise.resolve()
   .then(function() {
-    ui.log('info', 'If using two-factor authentication or to avoid using your password you can generate an access token at %https://' + (config.hostname || 'github.com') + '/settings/tokens%. Ensure it has `public_repo` scope access.');
-    return ui.input('Enter your GitHub username');
+    ui.log('info', 'You can generate an access token at %https://' + (config.hostname || 'github.com') + '/settings/tokens%.');
+    return ui.input('Enter your GitHub access token');
   })
-  .then(function(username) {
-    auth.username = username;
-    if (auth.username)
-      return ui.input('Enter your GitHub password or access token', null, true);
-  })
-  .then(function(password) {
-    auth.password = password;
-    if (!auth.username)
-      return false;
-
-    return ui.confirm('Would you like to test these credentials?', true);
```

Comment:
Because the prompts form part of the public API.

If the PR was against the 0.17 branch for the jspm beta then we can
definitely consider doing that for 0.17. But on the 0.16 branch they should
remain as users expect now to be able to directly enter their github
username and password (which allows not having to go to the website to
create a token at all)
On Sun, 12 Jun 2016 at 02:54, Tamir Duberstein <notifications@github.com>
wrote:

> In github.js <https://github.com/jspm/github/pull/90#discussion_r66714150>
> :
>
> >    })
> > -  .then(function(username) {
> > -    auth.username = username;
> > -    if (auth.username)
> > -      return ui.input('Enter your GitHub password or access token', null, true);
> > -  })
> > -  .then(function(password) {
> > -    auth.password = password;
> > -    if (!auth.username)
> > -      return false;
> > -
> > -    return ui.confirm('Would you like to test these credentials?', true);
>
> Why? That means none of this can ever be deprecated or removed. We want to
> make sure people's existing configuration works, but why does that mean we
> need to support both forms forever?
>
> That also creates a more confusing situation with the access tokens - if
> you use username/password you need to give public_repo, and otherwise you
> don't. Doesn't seem great
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/jspm/github/pull/90/files/7e73f5e239dac1f8eca852495dc8029f82991f26#r66714150>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAkiykjKrj7oHuNSqxdXtl8EFq_VgRjXks5qK1izgaJpZM4Izqxv>
> .
>